src/_imagingft.c

```diff
@@ -673,15 +950,15 @@ font_getsize(FontObject *self, PyObject *args) {
             }
             switch (anchor[1]) {
                 case 'a':  // ascender
-                    y_anchor = PIXEL(self->face->size->metrics.ascender);
+                    y_anchor = PIXEL(family->faces[0]->size->metrics.ascender);
                     break;
                 case 't':  // top
                     y_anchor = y_max;
                     break;
                 case 'm':  // middle (ascender + descender) / 2
                     y_anchor = PIXEL(
-                        (self->face->size->metrics.ascender +
-                         self->face->size->metrics.descender) /
+                        (family->faces[0]->size->metrics.ascender +
+                         family->faces[0]->size->metrics.descender) /
                         2);
                     break;
                 case 's':  // horizontal baseline
```

Comment:
I think here you would need to get the ascenders and descenders of every font used on this line, not just the first one, and use the greatest value.

Reply:
Perhaps. I don't think looking just at the fonts used in a given line would be good - that could create an inconsistent layout for the `text_multiline` functions.

This should use either the first font's metrics or the greatest value across all fonts in a family object, and it should be consistent with `family.getmetrics()` (not yet implemented).

Reply:
I've just tested what MS Edge (i.e. Chromium) does - it uses the metrics of all fonts up to the last one that is in use for a given line, even if it causes uneven line spacing.

For example, with three fonts, a paragraph that only uses the second font will have line spacing computed from the first two fonts. If one word in the paragraph then uses the third font, that line will include the third font's line spacing, but the other lines are unaffected.

This is incompatible with how Pillow currently calculates line spacing for multiline text (see https://github.com/python-pillow/Pillow/issues/6469#issuecomment-1203036583), so I think I'll just use the maximum over all fonts (i.e. assume that if a user creates a `FontFamily` object they actually want to use all of the fonts), and leave fixing it for #1646 (+document the limitation).